library/blob.c

```diff
@@ -1482,6 +1507,21 @@ static int eblob_fill_write_control_from_ram(struct eblob_backend *b, struct ebl
 		eblob_dump_wc(b, key, wc, "eblob_fill_write_control_from_ram: ERROR-pread-index", err);
 		goto err_out_cleanup_wc;
 	}
+
+	err = __eblob_read_ll(wc->data_fd, &data_dc, sizeof(data_dc), ctl.data_offset);
+	if (err) {
+		eblob_dump_wc(b, key, wc, "eblob_fill_write_control_from_ram: ERROR-pread-data", err);
+		goto err_out_cleanup_wc;
+	}
+
+	/* mark entry removed if its headers from index and data are different */
+	if (memcmp(&dc, &data_dc, sizeof(dc))) {
+		err = -EINVAL;
```

Comment:
If we get such logs we can investigate the cause of the problem without losing such keys. It would be useful at the first stage while we don't know all reasons of the problem - maybe we have some bug which leads to headers inconsistent and this inconsistent can be easily overcome, but instead eblob will remove keys.

We can start to remove such keys, but in this case we have to have some util that will scan all our blobs and finds keys with inconsistent headers. Before running new version which will removes such keys, we should run the util and investigate all found keys.